src/opm/output/eclipse/AggregateGroupData.cpp

```diff
@@ -112,25 +112,13 @@ void groupLoop(const std::vector<const Opm::Group*>& groups,
     }
 }
 
-template < typename T>
-std::pair<bool, int > findInVector(const std::vector<T>  & vecOfElements, const T  & element)
+template <typename T>
+std::optional<int> findInVector(const std::vector<T>  & vecOfElements, const T  & element)
 {
-    std::pair<bool, int > result;
-
     // Find given element in vector
     auto it = std::find(vecOfElements.begin(), vecOfElements.end(), element);
 
-    if (it != vecOfElements.end())
-    {
-        result.second = std::distance(vecOfElements.begin(), it);
-        result.first = true;
-    }
-    else
-    {
-        result.first = false;
-        result.second = -1;
-    }
-    return result;
+    return (it != vecOfElements.end()) ? std::optional<int>{std::distance(vecOfElements.begin(), it)} : std::nullopt;
```

Comment:
You do not have to explicitly construct the `std::optional<>` - i.e. this would also work:
```
if (it != vecOfElements.end()) ? std::distance(vecOfElements.begin(), it) : {};
```

Reply:
I agree to simplify - omitting the optional <> construction. However, I keep the std::nullopt - which is a standard to use for an std::optional construction.